### ISSUE TITLE
pio library not being found

### ISSUE THREAD

**Issue by neil9876** [EarthWorksOrg/EarthWorks#88]
When I run EarthWorks, it aborts with the following error in cesm.log:

```
/project/6024112/ntandon/earthworks2_3/EarthWorks_F2000_120ao/bld/cesm.exe: error while loadin
g shared libraries: libpiof.so.4: cannot open shared object file: No such file or directory

Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.

mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[10316,1],0]
  Exit code:    127

```

I built PIO 2.6.2 myself and my config machines.xml contains the following PIO-related statements:

`      <env name="PIO_VERSION_MAJOR">2</env>
      <env name="PIO_LIBDIR">/project/6024112/ntandon/software/StdEnv2023_nvhpc23.9_openmpi4.\
1.5/pio-2.6.2/lib</env>
      <env name="PIO_INCDIR">/project/6024112/ntandon/software/StdEnv2023_nvhpc23.9_openmpi4.\
1.5/pio-2.6.2/include</env>
      <env name="PIO_TYPENAME_VALID_VALUES">netcdf, pnetcdf, netcdf4c, netcdf4p</env>`

These paths are correct, and  libpiof.so.4 is contained in the indicated PIO_LIBDIR. So is there some other environment variable that I need to set for cesm to properly link to my PIO library?

**Comment by gdicker1** [comment]
My suspicion is that something isn't set up correctly in your config_machines.xml and your `PIO_...` variables are in your environment but not added to your `PATH` or `LD_LIBRARY_PATH` when you run `./case.submit`

From the config_machines.xml that you shared in #85, I think you're missing something that modifies `PATH` and `LD_LIBRARY_PATH`. You may be able to add something like the below:

```xml
<!-- Your existing entry -->
     <environment_variables compiler="nvhpc" mpilib="openmpi">
       <!-- Lines that already exist in your config being skipped... -->
       <!-- Vars you may need to add (PIO_BINDIR is just for illustration, you may need to do something similar for NETCDF_PATH instead) -->
       <env name="PIO_BINDIR">/VALID/PATH/TO/bin<env>
       <env name="PIO_LIBDIR">/project/6024112/ntandon/software/StdEnv2023_nvhpc23.9_openmpi4.\ 1.5/pio-2.6.2/lib</env>
       <!-- Now extended to modify PATH and LD_LIBRARY_PATH, you may need to insert all your variables above too -->
       <env name="PATH">${PATH}:${PIO_BINDIR}"</env>
       <env name="LD_LIBRARY_PATH">${LD_LIBRARY_PATH}:${PIO_LIBDIR}</env>
     </environment_variables>
```

(This is just my suggestion right now, I haven't tried getting EarthWorks or CESM to use a custom built library before.)

**Comment by neil9876** [comment]
Ah ok, I think when the system has these packages available as loadable modules, these variables get set automatically. But if I build these packages myself then I have to set these things. In any case, I added statements to config_machines.xml to set the PATH and LD_LIBRARY_PATH, and I also appended library paths to SLIBS in the cmake macro file (my_earthworks_sandbox/ccs_config/machines/cmake_macros/nvhpc_narval.cmake), and I don't get the error anymore.

For those used to CESM2, it appears that the cmake macro files have replace the config_compilers.xml file.
